### ISSUE TITLE
TeamCity does not interrupt NUnit test run

### ISSUE THREAD

**Issue by YRudievGP** [JetBrains/teamcity-dotnet-plugin#125]
Tests are not stopped if cancel TeamCity run.

Steps to reproduce:

Run TeamCity test run with NUnit tests
When tests are started executing, cancel test run (press " stop build" button)
Result:
Tests are continued executing.

Additional information:
It is actually for UI and backend tests.

In use:
.Net framework netcoreapp2.0
NUnit 3.10.1
NUnit.ConsoleRunner 3.8.0
NUnit.Runners 3.8.0
NUnit3TestAdapter 3.10.0
Runner type .NET CLI (dotnet)

According to TeamCity logs, when "stop build" button was pressed, docker-composeer was stopped but tests are still running.

Part of logs from TeamCity:
Step 10/11: Run test by Dotnet (.NET CLI (dotnet)) (1m:01s)
[12:21:50][Step 10/11] dotnet test ServerTesting/Server.Testing.SystemTests/Server.Testing.SystemTests.csproj --filter "TestCategory=bingo|TestCategory=common|TestCategory=nolobby" --configuration CI @/opt/TeamCity/BuildAgent1/temp/agentTmp/7bae7bbb359b4441ab48c293510bddc7.rsp
[12:21:50][Step 10/11] test (59s)
[12:21:50][test] Starting: /usr/bin/dotnet test ServerTesting/Server.Testing.SystemTests/Server.Testing.SystemTests.csproj --filter "TestCategory=bingo|TestCategory=common|TestCategory=nolobby" --configuration CI @/opt/TeamCity/BuildAgent1/temp/agentTmp/7bae7bbb359b4441ab48c293510bddc7.rsp
[12:21:50][test] in directory: /opt/TeamCity/BuildAgent1/work/668a102bb75fdb62
[12:21:52][test] Сборка начата, подождите...
[12:21:56][test] Сборка выполнена.
[12:21:56][test]
[12:21:56][test] Тестовый запуск /opt/TeamCity/BuildAgent1/work/668a102bb75fdb62/ServerTesting/Server.Testing.SystemTests/bin/CI/netcoreapp2.0/Server.Testing.SystemTests.dll(.NETCoreApp,Version=v2.0)
[12:21:56][test] Программа Microsoft (R) Test Execution Command Line Tool версии 15.6.0
[12:21:56][test] (с) Корпорация Майкрософт (Microsoft Corporation). Все права защищены.
[12:21:56][test]
[12:21:56][test] Запуск выполнения тестов; подождите...
[12:21:57][test] NUnit Adapter 3.10.0.21: Test execution started
[12:21:57][test] Running all tests in /opt/TeamCity/BuildAgent1/work/668a102bb75fdb62/ServerTesting/Server.Testing.SystemTests/bin/CI/netcoreapp2.0/Server.Testing.SystemTests.dll
[12:21:57][test] NUnit3TestExecutor converted 198 of 198 NUnit test cases
[12:22:00][test] Server.Testing.SystemTests.dll (49s)
[12:22:00][Server.Testing.SystemTests.dll] Server.Testing.SystemTests.Common.BasicStateCommon.Buddy.CAckAddBuddyTests.CAckAddBuddy_LimitOfBuddiesReached_SDeniedAddingBuddyReceived
[12:22:03][Server.Testing.SystemTests.dll] Server.Testing.SystemTests.Common.BasicStateCommon.Buddy.CAckAddBuddyTests.CAckAddBuddy_Negative_BuddyAuthorizationDeclined
[12:22:05][Server.Testing.SystemTests.dll] Server.Testing.SystemTests.Common.BasicStateCommon.Buddy.CAckAddBuddyTests.CAckAddBuddy_Negative_RelationDublicateIsDeleted
[12:22:08][Server.Testing.SystemTests.dll] Server.Testing.SystemTests.Common.BasicStateCommon.Buddy.CAckAddBuddyTests.CAckAddBuddy_Positive_BuddyAuthorizationGranted
[12:22:10][Server.Testing.SystemTests.dll] Server.Testing.SystemTests.Common.BasicStateCommon.Buddy.CAckAddBuddyTests.CAckAddBuddy_Positive_RelationDublicateIsDeleted
[12:22:13][Server.Testing.SystemTests.dll] Server.Testing.SystemTests.Common.BasicStateCommon.Buddy.CAckAddBuddyTests.CAckAddBuddy_RelationInAuthorizationRequired2_BuddyAuthorizationForbidden
[12:22:14][test] Stopping docker-compose for docker-compose.yml
[12:22:22][Server.Testing.SystemTests.dll] Server.Testing.SystemTests.Common.BasicStateCommon.Buddy.CAckAddBuddyTests.CAckAddBuddy_RelationInAuthorizationRequired2_BuddyAuthorizationGranted
[12:22:49][test] Не пройдено: CAddBuddyTests_BuddyAuthorizationInvalid_RelationAuthorizationRequiredAndSAddBuddyReceived

If I can provide any additional information, please let me know.

**Comment by NikolayPianikov** [comment]
Fixed in TeamCity 2017.2.x and master
@YRudievGP try updating your plugin

**Comment by YRudievGP** [comment]
Currently in use TeamCity 2017.2.1 and issue is reproduced
May try to update higher, if it might help.

**Comment by NikolayPianikov** [comment]
@YRudievGP  Have you updated dotnet CLI plugin from [this](https://github.com/JetBrains/teamcity-dotnet-plugin#download) page?

**Comment by YRudievGP** [comment]
No, I haven't. As it is bundled with TeamCity, it should have been updated automatically with TeamCity.
We will update TeamCity to 2017.2.4 (build 51228) this night and it might get latest version of dotnet plugin. 

Will reply if it helps

**Comment by NikolayPianikov** [comment]
I've fixed it not so long ago (just several hours) so it is not in any releases yet. You might update this plugin manually from the suggested page (it contains an instruction) or could wait for next TeamCity release if it is not so urgent

**Comment by YRudievGP** [comment]
Thanks for the clarification

**Comment by YRudievGP** [comment]
We have added plugin manually and problem was solved. Thanks
